internal/controllers/core/clustercatalog_controller.go

```diff
@@ -180,73 +180,60 @@ func (r *ClusterCatalogReconciler) reconcile(ctx context.Context, catalog *v1alp
 
 		return ctrl.Result{RequeueAfter: requeueAfter}, nil
 	default:
-		return ctrl.Result{}, updateStatusUnpackFailing(&catalog.Status, fmt.Errorf("unknown unpack state %q: %v", unpackResult.State, err))
+		panic(fmt.Sprintf("unknown unpack state %q", unpackResult.State))
 	}
 }
 
-func updateStatusUnpackPending(status *v1alpha1.ClusterCatalogStatus, result *source.Result) {
-	status.ResolvedSource = nil
-	meta.SetStatusCondition(&status.Conditions, metav1.Condition{
-		Type:    v1alpha1.TypeUnpacked,
-		Status:  metav1.ConditionFalse,
-		Reason:  v1alpha1.ReasonUnpackPending,
-		Message: result.Message,
-	})
-}
+func updateStatusProgressing(catalog *v1alpha1.ClusterCatalog, err error) {
```

Comment:
```suggestion
func updateStatusProgressing(*v1alpha1.ClusterCatalog, error) {
```

Use explicit returns: https://go.dev/tour/basics/7

Reply:
would it be worth refactoring the package name to `catd` or `catdv1alpha1`?

Reply:
Just a note that these look to be input parameters to the function, not return values

Reply:
Ah I see why @joelanford may have thought that these were named return values. We are doing "naked returns" in this function. To reduce potential for confusion, maybe we should do something like:
```go
func updateStatusProgression(catalog *v1alpha1.ClusterCatalog, err error) {
        progressingCond := metav1.Condition{
                Type:  v1alpha1.TypeProgressing,
                Status: metav1.ConditionFalse,
                Reason: v1alpha1.ReasonSucceeded,
                Message: "...",
       }
       
       if err != nil {
               progressingCond.Status = metav1.ConditionTrue
               progressingCond.Reason = v1alpha1.ReasonRetrying
               progressingCond.Message = err.Error()
       }
       
       if errors.As(err, &unrecov) {
               progressingCond.Status = metav1.ConditionFalse
               progressingCond.Reason = v1alpha1.ReasonUnrecoverable
       }
       
       meta.SetStatusCondition(&catalog.Status.Conditions, progressingCond)
}
```

Reply:
Oh yeah, @everettraven is right. I saw the naked returns and the error parameter and my mind jumped to named returns. My bad!